src/CrudTrait.php

```diff
@@ -191,4 +191,215 @@ public function uploadMultipleFilesToDisk($value, $attribute_name, $disk, $desti
+
+                    $variant_name = $new_file_name.'-'.$variant.'.'.$file->getClientOriginalExtension();
+                    $variant_file = $destination_path.'/'.$variant_name;
+
+                    if ($dimensions) {
+                        $width = $dimensions[0];
+                        $height = $dimensions[1];
+
+                        if ($img->width() > $width || $img->height() > $height) {
+                            $img->resize($width, $height, function ($constraint) {
+                                $constraint->aspectRatio();
+                            })
+                            ->save($disk_root.'/'.$variant_file);
+                        } else {
+                            $img->save($disk_root.'/'.$variant_file);
+                        }
+
+                        $image_variations[$variant] = $public_path.'/'.$variant_file;
+                    } else {
+                        $image_variations['original'] = $public_path.'/'.$file_path;
+                    }
+                }
+            } else {
+                $image_variations['original'] = $public_path.'/'.$file_path;
+                $image_variations['thumb'] = $public_path.'/'.$file_path;
+            }
+
+            // 3. Save the complete path to the database
+            $this->attributes[$attribute_name] = $image_variations['original'];
+        } elseif (starts_with($value, 'data:image')) {
+            $new_file_name = md5($value.time());
+
+            if (! \Illuminate\Support\Facades\File::exists($disk_root.'/'.trim($destination_path, '/'))) {
+                \Illuminate\Support\Facades\File::makeDirectory($disk_root.'/'.trim($destination_path, '/'), 0775, true);
+            }
+
+            foreach ($variations as $variant => $dimensions) {
+                $img = \Intervention\Image\ImageManagerStatic::make($value);
+
+                switch ($img->mime()) {
+                    case 'image/bmp':
+                    case 'image/ief':
+                    case 'image/jpeg':
+                    case 'image/pipeg':
+                    case 'image/tiff':
+                    case 'image/x-jps':
+                        $extension = '.jpg';
+                        break;
+                    case 'image/gif':
+                        $extension = '.gif';
+                        break;
+                    case 'image/x-icon':
+                    case 'image/png':
+                        $extension = '.png';
+                        break;
+                    default:
+                        $extension = '.jpg';
+                        break;
+                }
+
+                $variant_name = $new_file_name.'-'.$variant.$extension;
+                $variant_file = $destination_path.'/'.$variant_name;
+
+                if ($dimensions) {
+                    $width = $dimensions[0];
+                    $height = $dimensions[1];
+
+                    if ($img->width() > $width || $img->height() > $height) {
```

Comment:
I'm sure I saw this same clause up there (see up there). Plus it's a possible candidate for DRY.